plugins/generator-1.15.2/forge-1.15.2/procedures/_set_variable.java.ftl

```diff
@@ -72,13 +72,13 @@
     }
     </#if>
 <#elseif scope == "local">
-    <#if type == "NUMBER">
+    <#if type == "number">
         ${name} =(double)${value};
-    <#elseif type == "LOGIC">
+    <#elseif type == "logic">
         ${name} =(boolean)${value};
-    <#elseif type == "STRING">
+    <#elseif type == "string">
         ${name} =(String)${value};
-    <#elseif type == "ITEMSTACK">
+    <#elseif type == "itemstack">
         ${name} = ${mappedMCItemToItemStackCode(value, 1)};
     </#if>

```

Comment:
The "Set variable" block doesn't actually generate any code for other variable types (same for 1.16)

Reply:
Keeping unresolved so I don't forget to test this

Reply:
Yup this works now :)